src/main/java/com/unstoppabledomains/resolution/ResolutionBuilder.java

```diff
@@ -0,0 +1,242 @@
+     */
+    public ResolutionBuilder infura(NamingServiceType nsType, String projectId) {
+        NSConfig nsConfig = serviceConfigs.get(nsType);
+        final Network network = nsConfig.getChainId();
+        return this.infura(nsConfig, network, projectId);
+    }
+
+    /**
+     * Configuration for <a href="https://infura.io">infura.io</a>
+     * blockchain provider with previously set network ID using
+     * {@link ResolutionBuilder#chainId(NamingServiceType, Network)} or default one
+     * ({@link Network#MAINNET})
+     *
+     * @param location the location of the UNS service (layer 1 or layer 2)
+     * @param projectId Infura project ID
+     * @return builder object to allow chaining
+     */
+    public ResolutionBuilder infura(UNSLocation location, String projectId) {
+        NSConfig nsConfig = unsConfigs.get(location);
+        final Network network = nsConfig.getChainId();
+        return this.infura(nsConfig, network, projectId);
+    }
+
+    /**
+     * Configuration for <a href="https://infura.io">infura</a> blockchain provider
+     *
+     * @param nsType    the naming service for which config is applied
+     * @param chainId   blockchain network ID
+     * @param projectId Infura project ID
+     * @return builder object to allow chaining
+     */
+    public ResolutionBuilder infura(NamingServiceType nsType, Network chainId, String projectId) {
+        NSConfig nsConfig = serviceConfigs.get(nsType);
+        return this.infura(nsConfig, chainId, projectId);
+    }
+
+    /**
+     * Configuration for <a href="https://infura.io">infura</a> blockchain provider
+     *
+     * @param location the location of the UNS service (layer 1 or layer 2)
+     * @param chainId   blockchain network ID
+     * @param projectId Infura project ID
+     * @return builder object to allow chaining
+     */
+    public ResolutionBuilder infura(UNSLocation location, Network chainId, String projectId) {
+        NSConfig nsConfig = unsConfigs.get(location);
+        return this.infura(nsConfig, chainId, projectId);
+    }
+
+    private ResolutionBuilder infura(NSConfig nsConfig, Network chainId, String projectId) {
+        nsConfig.setChainId(chainId);
+
+        String infuraUrl = String.format(INFURA_URL, chainId.getName(), projectId);
+        nsConfig.setBlockchainProviderUrl(infuraUrl);
+
+        return this;
+    }
+
+    /**
+     * Configuration for provider interface, use if needed to take control over network calls
+     * @param provider must implement IProvider interface
+     * @return builder object to allow chaining
+     */
+    public ResolutionBuilder provider(IProvider provider) {
+        setProvider(provider);
+        return this;
+    } 
+
+    /**
+     * Call directly to get default configs or override them with {@link ResolutionBuilder} methods
+     *
+     * @return resolution object
+     */
+    public Resolution build() {
+        Map<NamingServiceType, NamingService> services = new HashMap<>();
```

Comment:
For the first check, we have defaults for L1 and L2. So if the users sets only one of them, the other will remain as default. So there should be no problem in this case. Or am I missing something?

And the second check is relevant only for UNS, or for ZNS and ENS too?

Reply:
@DeRain I've added the second condition. Not sure about the first one still.

Reply:
@DancingAxolotl We're trying to avoid case when user sets L1 testnet and L2 remains mainnet by default or vice versa. So we ask user explicitly set L1 and L2 together and user will be responsible for setting both network correctly.